lib/riemann.js

```diff
@@ -18,12 +39,11 @@ function Riemann(startupTime, config, emitter) {
 
   // Bind some Riemann handlers
   this.client.once("connect", u.bind(this.onConnect, this));
-  this.client.once("error", u.bind(this.onError, this));
+  this.client.on("error", u.bind(this.onError, this));
   this.client.on("data", u.bind(this.onAck, this));
 
-  // Finally, bind to StatsD eventemitter
-  emitter.on("packet", u.bind(this.onPacket, this));
-}
+  this.emitter.on("packet", u.bind(this.onPacket, this));
```

Comment:
I don't think you need to register events on the emitter itself on each error. This will likely double register the event handlers since the emitter is not disposed during a Riemann-level error. I would just add the emitter registration in the constructor function and leave the client event registration bits here since those will need to be re-registered each time the client is constructed.